src/game/logic/system/gamelogic.cpp

```diff
@@ -390,3 +609,2356 @@ GhostObjectManager *GameLogic::Create_Ghost_Object_Manager()
+        "(Top "
+        "is '%s')",
+        Get_First_Object()->Get_Template()->Get_Name().Str());
+}
+
+void GameLogic::Process_Destroy_List()
+{
+    for (auto it = m_objectsToDestroy.begin(); it != m_objectsToDestroy.end(); it++) {
+        int count = 0;
+        UpdateModule *updates[256];
+        Object *obj = (*it);
+
+        for (auto it2 = m_sleepingUpdateModules.begin(); it2 != m_sleepingUpdateModules.end(); it2++) {
+            UpdateModule *update = *it2;
+
+            if (update->Get_Object() == obj && count < 256) {
+                updates[count++] = update;
+            }
+        }
+
+        count--;
+
+        while (count >= 0) {
+            int index = updates[count]->Get_Index_In_Logic();
+            captainslog_dbgassert(m_sleepingUpdateModules[index] == updates[count], "Hmm, expected update mismatch here");
+            Erase_Sleepy_Update(index);
+            captainslog_dbgassert(updates[count]->Get_Index_In_Logic() == -1, "Hmm, expected index to be -1 here");
+            --count;
+        }
+
+        obj->Remove_From_List(&m_objList);
+        Remove_Object_From_Lookup_Table(obj);
+        obj->Delete();
+    }
+
+    m_objectsToDestroy.clear();
+}
+
+void GameLogic::Erase_Sleepy_Update(int index)
+{
+    captainslog_dbgassert(index >= 0 && index < static_cast<int>(m_sleepingUpdateModules.size()), "bad sleepy idx");
+    m_sleepingUpdateModules[index]->Set_Index_In_Logic(-1);
+    int last_index = m_sleepingUpdateModules.size() - 1;
+
+    if (index >= last_index) {
+        m_sleepingUpdateModules.pop_back();
+    } else {
+        m_sleepingUpdateModules[index] = m_sleepingUpdateModules[last_index];
+        m_sleepingUpdateModules[index]->Set_Index_In_Logic(index);
+        m_sleepingUpdateModules.pop_back();
+        Rebalance_Sleepy_Update(index);
+    }
+}
+
+void GameLogic::Remake_Sleepy_Update()
+{
+    for (unsigned int i = m_sleepingUpdateModules.size() >> 1;; i--) {
+        Rebalance_Child_Sleepy_Update(i);
+
+        if (i == 0) {
+            break;
+        }
+    }
+}
+
+void GameLogic::Xfer_Object_TOC(Xfer *xfer)
+{
+    unsigned char version = 1;
+    xfer->xferVersion(&version, 1);
+    m_objectTOCEntries.clear();
+    unsigned int count = 0;
+
+    if (xfer->Get_Mode() == XFER_LOAD) {
+        Utf8String name;
+
+        for (Object *obj = Get_First_Object(); obj != nullptr; obj = obj->Get_Next_Object()) {
+            name = obj->Get_Template()->Get_Name();
+
+            if (!Find_TOC_Entry_By_Name(name)) {
+                Add_TOC_Entry(obj->Get_Template()->Get_Name(), ++count);
+            }
+        }
+
+        xfer->xferUnsignedInt(&count);
+
+        for (auto it = m_objectTOCEntries.begin(); it != m_objectTOCEntries.end(); it++) {
+            xfer->xferAsciiString(&(*it).name);
+            xfer->xferUnsignedShort(&(*it).id);
+        }
+    } else {
+        Utf8String name;
+        unsigned short id;
+        unsigned int count2;
+        xfer->xferUnsignedInt(&count2);

```

Comment:
This does not look right. It will save random value here.

Reply:
fixed.